src/commands/check.rs

```diff
@@ -1 +1,113 @@
+use glob::Pattern;
+use std::fs::File;
+use std::io::BufRead;
+use std::io::BufReader;
+use std::path::Path;
+use walkdir::WalkDir;
 
+use super::common::exit_codes;
+use super::common::format_message_line;
+use super::common::DirectoryCommand;
+use super::error::Error;
+use crate::color::palette::Palette;
+use crate::config::data::Workspace;
+
+pub struct Check {}
+
+impl DirectoryCommand for Check {
+    fn run(
+        &self,
+        working_dir: &Path,
+        workspace: &Workspace,
+        palette: &Palette,
+    ) -> Result<i32, Error> {
+        let ignored_dirs = read_ignore_file(working_dir);
+        let projects = check_projects(working_dir, workspace, &ignored_dirs);
+
+        for project in projects.iter() {
+            println!("{}:", palette.repo.paint(&project.name));
+            match project.status {
+                ProjectStatus::Ignored => {
+                    println!("{}", palette.clean.paint(format_message_line("Ignored")));
+                }
+                ProjectStatus::Known => {
+                    println!("{}", palette.clean.paint(format_message_line("Known")));
+                }
+                ProjectStatus::Missing => {
+                    println!("{}", palette.error.paint(format_message_line("Missing")));
+                }
+                ProjectStatus::Unknown => {
+                    println!("{}", palette.error.paint(format_message_line("Unknown")));
+                }
+            }
+        }
+
+        Ok(exit_codes::OK)
+    }
+}
+
+struct CheckProject {
+    name: String,
+    status: ProjectStatus,
+}
+
+enum ProjectStatus {
+    Known,
+    Missing,
+    Unknown,
+    Ignored,
+}
+
+fn read_ignore_file(dir: &Path) -> Vec<Pattern> {
+    let file = File::open(dir.join(".ignore.gws")).unwrap();
+    let reader = BufReader::new(file);
+    reader
+        .lines()
+        .filter_map(Result::ok)
+        .filter(|line| !line.starts_with("#"))
+        .filter_map(|line| Pattern::new(&format!("{}/**", line.trim())).ok())
```

Comment:
the globbing isn't working just yet.

e.g. ignoring

`.terraform/` still includes folders like this

```
terraform-examples/environments/bar/.terraform/modules/cluster.aws_ebs_csi_driver_irsa
terraform-examples/environments/bar/.terraform/modules/cluster.eks
terraform-examples/environments/foo/.terraform/modules/cluster.aws_ebs_csi_driver_irsa
terraform-examples/environments/foo/.terraform/modules/cluster.eks
```